settings/resource_notification_destination.go

```diff
@@ -0,0 +1,177 @@
+package settings
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func setStruct(s *settings.NotificationDestination, readND *settings.NotificationDestination) {
+	switch readND.DestinationType {
+	case settings.DestinationTypeSlack:
+		readND.Config.Slack.Url = s.Config.Slack.Url
+	case settings.DestinationTypePagerduty:
+		readND.Config.Pagerduty.IntegrationKey = s.Config.Pagerduty.IntegrationKey
+	case settings.DestinationTypeMicrosoftTeams:
+		readND.Config.MicrosoftTeams.Url = s.Config.MicrosoftTeams.Url
+	case settings.DestinationTypeWebhook:
+		if readND.Config.GenericWebhook.UrlSet {
+			readND.Config.GenericWebhook.Url = s.Config.GenericWebhook.Url
+		}
+		if readND.Config.GenericWebhook.PasswordSet {
+			readND.Config.GenericWebhook.Password = s.Config.GenericWebhook.Password
+		}
+		if readND.Config.GenericWebhook.UsernameSet {
+			readND.Config.GenericWebhook.Username = s.Config.GenericWebhook.Username
+		}
+	}
+}
+
+func Create(ctx context.Context, d *schema.ResourceData, w *databricks.WorkspaceClient) error {
+	var newNDrequest settings.CreateNotificationDestinationRequest
+	common.DataToStructPointer(d, ndSchema, &newNDrequest)
+	createdND, err := w.NotificationDestinations.Create(ctx, newNDrequest)
+	if err != nil {
+		return err
+	}
+	d.SetId(createdND.Id)
+	d.Set("destination_type", createdND.DestinationType)
+	return nil
+}
+
+func Read(ctx context.Context, d *schema.ResourceData, w *databricks.WorkspaceClient) error {
+	var tempND settings.NotificationDestination
+	common.DataToStructPointer(d, ndSchema, &tempND)
+
+	readND, err := w.NotificationDestinations.Get(ctx, settings.GetNotificationDestinationRequest{
+		Id: d.Id(),
+	})
+	if err != nil {
+		return err
+	}
+	setStruct(&tempND, readND)
+	return common.StructToData(readND, ndSchema, d)
+}
+
+func detectConfigTypeChange(d *schema.ResourceData) bool {
+	switch d.Get("destination_type").(string) {
+	case string(settings.DestinationTypeSlack):
+		_, ok := d.GetOk("config.0.slack")
+		return !ok
+	case string(settings.DestinationTypePagerduty):
+		_, ok := d.GetOk("config.0.pagerduty")
+		return !ok
+	case string(settings.DestinationTypeMicrosoftTeams):
+		_, ok := d.GetOk("config.0.microsoft_teams")
+		return !ok
+	case string(settings.DestinationTypeWebhook):
+		_, ok := d.GetOk("config.0.generic_webhook")
+		return !ok
+	}
+	return false
+}
+
+func Update(ctx context.Context, d *schema.ResourceData, w *databricks.WorkspaceClient) error {
+	if detectConfigTypeChange(d) {
+		err := Delete(ctx, d, w)
+		if err != nil {
+			return err
+		}
+		return Create(ctx, d, w)
+	}
```

Comment:
According to API it should be possible to update it inline: https://docs.databricks.com/api/workspace/notificationdestinations/update - the main problem is that if we do delete/create, then ID of destination will change and this may break other integrations

Reply:
That is true, but the update API does not allow to change the type of notification destination. I created an integration test `TestAccConfigTypeChange` for this exact puspose. If we do not delete and create a new resouce and try to update  it instead, the update API response is `bad request` and states that you can't change config types using PATCH call

Reply:
We need to raise an internal ticket to update the OpenAPI spec to reflect this - right now, it's not documented.  But it's better first to confirm that this is intended behavior. 

Instead of implicitly recreating the resource, we may need to mark these blocks as `force_new`, so Terraform will delete/create automatically. We need to do this because Terraform needs to know that `id` is changing, and this change is propagated to the dependent resources. Otherwise, they could be broken after such an update. 

We also need to document that behavior.

Reply:
1) which team should I contact for this.
2) I will try to make fields force new rather than explicitly doing it.

Reply:
Also, is raising the ticket a blocking change for this PR?